### PR TITLE
Updating FileRecordStream docstrings

### LINKED ISSUE #3542: Update FileRecordStream API docs

### PR DESCRIPTION
**Pull request by rhyolight**
Fixes #3542

Blocked by #3541

### CODE REVIEW

File: src/nupic/data/file_record_stream.py [hunk 1]
```diff
@@ -22,43 +22,55 @@
 """
 CSV file based implementation of a record stream
 
-FileRecordStream is class that can read and write .csv files that contain
-records. The file has 3 header lines that contain, for each field, the name
-(line 1), type (line 2), and a special indicator (line 3). The special indicator
-can indicate that the field specifies a reset, is a sequence ID, or is a
-timestamp for the record.
+:class:`~.file_record_stream.FileRecordStream` is class that can read and write
```
Comment (marionleborgne) [inline src/nupic/data/file_record_stream.py]: `FileRecordStream` is *a* class

File: src/nupic/data/file_record_stream.py [hunk 2]
```diff
@@ -22,43 +22,55 @@
...
 The 'us' component is microseconds.
 
 When reading a file the FileRecordStream will automatically read the header line
 and will figure out the type of each field and what are the timestamp, reset
 and sequenceId fields (if any).
 
-The FileRecordStream class supports the context manager ('with' statement )
-protocol. That means you con do:
+:class:`~.file_record_stream.FileRecordStream` supports the context manager
+(``with`` statement) protocol. That means you con do:
```
Comment (marionleborgne) [inline src/nupic/data/file_record_stream.py]: Typo: You can do

File: src/nupic/data/file_record_stream.py [hunk 3]
```diff
@@ -324,9 +343,10 @@ def rewind(self):
   def getNextRecord(self, useCache=True):
     """ Returns next available data record from the file.
 
-    retval: a data row (a list or tuple) if available; None, if no more records
-             in the table (End of Stream - EOS); empty sequence (list or tuple)
-             when timing out while waiting for the next record.
+    :param useCache: NOT USED
```
Comment (marionleborgne) [inline src/nupic/data/file_record_stream.py]: Maybe we should just get rid of this?

File: src/nupic/data/file_record_stream.py [hunk 4]
```diff
@@ -395,15 +401,12 @@ def removeOldData(self):
...
-        record: a list of Python objects that will be string-ified
-
-        Returns: nothing
     """
+    Saves the record in the underlying csv file.
 
-    # input bookmark is not applicable in case of a file storage
-    inputBookmark = inputBookmark
+    :param record: a list of Python objects that will be string-ified
+    :param inputBookmark: NOT USED
```
Comment (marionleborgne) [inline src/nupic/data/file_record_stream.py]: Can we rid of this?

File: src/nupic/data/file_record_stream.py [hunk 5]
```diff
@@ -465,43 +470,61 @@ def getBookmark(self):
 
 
   def recordsExistAfter(self, bookmark):
-    """Returns True iff there are records left after the  bookmark."""
+    """
+    Returns whether there are more records from current position. ``bookmark``
+    is not used in this implementation.
+
+    :param bookmark: NOT USED
```
Comment (marionleborgne) [inline src/nupic/data/file_record_stream.py]: Same question as before, ca we remove it since it's not used.

**Comment by rhyolight** [comment]
@marionleborgne took all your feedback, thanks

**Pull request merged**